server/test/unit/server/webservices/views/test_consumer_groups.py

```diff
@@ -0,0 +1,296 @@
+        request = mock.MagicMock()
+        request.body_as_json = {'id': 'foo', 'display_name': 'bar'}
+        mock_factory.consumer_group_manager.return_value.create_consumer_group.return_value = resp
+        consumer_group = ConsumerGroupView()
+        response = consumer_group.post(request)
+
+        self.assertTrue(isinstance(response, HttpResponseRedirect))
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(response._headers.get('content-type'),
+                         ('Content-Type', 'application/json'))
+        content = json.loads(response.content)
+        self.assertEqual(content, expected_cont)
+
+
+class TestconsumerGroupResourceView(unittest.TestCase):
+    """
+    Test consumer groups resource view.
+    """
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_DELETE())
+    @mock.patch('pulp.server.webservices.views.consumer_groups.factory')
+    def test_delete_consumer_group_resource(self, mock_factory):
+        """
+        Test consumer group delete resource.
+        """
+        mock_group_manager = mock.MagicMock()
+        mock_factory.consumer_group_manager.return_value = mock_group_manager
+        mock_group_manager.delete_consumer_group.return_value = None
+
+        request = mock.MagicMock()
+        consumer_group_resource = ConsumerGroupResourceView()
+        response = consumer_group_resource.delete(request, 'test-group')
+
+        self.assertTrue(isinstance(response, HttpResponse))
+        self.assertFalse(isinstance(response, HttpResponseNotFound))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response._headers.get('content-type'),
+                         ('Content-Type', 'application/json'))
+        content = json.loads(response.content)
+        self.assertEqual(content, None)
+        mock_group_manager.delete_consumer_group.assert_called_once_with('test-group')
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch('pulp.server.webservices.views.consumer_groups.ConsumerGroup.get_collection')
+    def test_get_consumer_group_resource(self, mock_collection):
+        """
+        Test single consumer group retrieval.
+        """
+        consumer_mock = mock.MagicMock()
+        consumer_mock.find_one.return_value = {'id': 'foo'}
+        mock_collection.return_value = consumer_mock
+
+        request = mock.MagicMock()
+        consumer_group = ConsumerGroupResourceView()
+        response = consumer_group.get(request, 'foo')
+
+        self.assertTrue(isinstance(response, HttpResponse))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response._headers.get('content-type'),
+                         ('Content-Type', 'application/json'))
+        content = json.loads(response.content)
+        expected_cont = {'id': 'foo', '_href': '/v2/consumer_groups/foo/'}
```

Comment:
I think this PR should introduce and use one or two helper methods similar to the helper methods @asmacdo  introduced: `generate_json_response_with_pulp_encoder()` and `generate_json_response()`